src/sql/src/plan/statement/ddl.rs

```diff
@@ -4502,8 +4490,18 @@ fn plan_retain_history(
                 "internal error: unexpectedly zero".to_string(),
             )),
         }),
-        Some(duration) => Ok(duration.try_into()?),
-        None => Ok(CompactionWindow::DisableCompaction),
+        Some(duration) => {
+            if duration < DEFAULT_LOGICAL_COMPACTION_WINDOW_DURATION {
```

Comment:
what do you think about making a second constant `MINIMUM_LOGICAL_COMPACTION_WINDOW_DURATION`? I wouldn't want someone to change the default in the future and then have users unable to set a compaction duration less than this?

Reply:
I don't quite understand your second sentence. If merged as is, and then in the future we bump the default to 2s, then users can't set the compaction to less than 2s. That seems fine to me?

Reply:
> If merged as is, and then in the future we bump the default to 2s, then users can't set the compaction to less than 2s.

That's the scenario I want to avoid :) it's not bad if we change it to 2s, but say we bump it to 60s because everything in the future is disk backed, but a user wants their index as slim as possible so they want to reduce it to 1s but now they can't.

I'm probably being too defensive here, so I'm good with whatever you choose!

Reply:
I'm going to leave it as is. If we bump it, all the tests will fail and someone will have to think about what to do, which sounds correct to me.